classes/view_form.php

```diff
@@ -1080,28 +1149,32 @@ public function add_readonly_browsing_buttons() {
             if ($maxassignedpage > 1) {
                 $formpage = $this->get_formpage();
                 if (($formpage != SURVEYPRO_LEFT_OVERFLOW) && ($formpage != 1)) {
-                    $params['formpage'] = $formpage - 1;
+                    $this->next_not_empty_page(false); // false means direction = left.
+                    $params['formpage'] = $this->get_nextpage();
                     $url = new \moodle_url('/mod/surveypro/view_form.php', $params);
-                    $backwardbutton = new \single_button($url, get_string('previousformpage', 'mod_surveypro'), 'get');
+                    $backwardbutton = new \single_button($url, get_string('previousformpage', 'mod_surveypro'), 'get', true);
                 }
 
                 if (($formpage != SURVEYPRO_RIGHT_OVERFLOW) && ($formpage != $maxassignedpage)) {
-                    $params['formpage'] = $formpage + 1;
+                    $this->next_not_empty_page(true); // true means direction = right.
+                    $params['formpage'] = $this->get_nextpage();
                     $url = new \moodle_url('/mod/surveypro/view_form.php', $params);
-                    $forwardbutton = new \single_button($url, get_string('nextformpage', 'mod_surveypro'), 'get');
+                    $forwardbutton = new \single_button($url, get_string('nextformpage', 'mod_surveypro'), 'get', true);
                 }
 
-                $params = array('class' => 'buttons');
+                $secondleveldiv = \html_writer::tag('div', '', ['class' => 'd-flex']);
+                $firstleveldiv = \html_writer::tag('div', $secondleveldiv, ['class' => 'col-md-4']);
                 if (isset($backwardbutton) && isset($forwardbutton)) {
-                    // This code comes from "public function confirm(" around line 1711 in outputrenderers.php.
-                    // It is not wrong. The misalign comes from bootstrapbase theme and is present in clean theme too.
-                    echo \html_writer::tag('div', $OUTPUT->render($backwardbutton).$OUTPUT->render($forwardbutton), $params);
+                    $content = $firstleveldiv.$OUTPUT->render($backwardbutton).$OUTPUT->render($forwardbutton);
+                    echo \html_writer::tag('div', $content, ['class' => 'row']);
                 } else {
                     if (isset($backwardbutton)) {
-                        echo \html_writer::tag('div', $OUTPUT->render($backwardbutton), $params);
+                        $content = $firstleveldiv.$OUTPUT->render($backwardbutton);
```

Comment:
Ciao Eloy.
I am trying to contact you on telegram but is seems you no longer receive my messages. Was something happening?
Daniele